shm/Ship.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>

```

Comment:
Cargo.hpp has string included,so you do not need it here.

Reply:
+1